main/authorization.bs

```diff
@@ -19,3 +21,31 @@ the HTTP `Link` header with a `rel` value of `acl` [[#acl]].
 In the event that a server can't apply an ACL to a resource, it MUST deny
 access.
 [[Source](https://github.com/solid/specification/issues/130#issue-532777017)]
+
+Servers exposing client's access privileges on a resource URL MUST advertise
+by including the `WAC-Allow` HTTP header in the response of HTTP `HEAD` and
+`GET` requests.
+
+The syntax for `WAC-Allow`, using the ABNF syntax is:
+
+```
+WAC-Allow           = "WAC-Allow" ":" OWS permissions OWS
+permissions         = user-permissions OWS "," OWS public-permissions
+user-permissions    = "user=" quoted-access-modes
+public-permissions  = "public=" quoted-access-modes
```

Comment:
What is the use case for including "public" access modes?

Reply:
Thanks for raising that. I forgot to mention that as well in the comment! The PR reflects the current issues and clarifies *solid-spec*.

If access modes for `user` can be determined for the current (un)authenticated agent, I don't see the use for `public`. Is there a UI that's currently using it or needs to differentiate between them? Some weird edge case where `user` has less privileges than `public`?

`user` doesn't need to be tied to authenticated agent either. On that note, spec needs to be more clear on what the parameter(s) are intended for.

I'd be in favour of dropping `public`. @RubenVerborgh @dmitrizagidulin

Reply:
That's a bit of a stretch no? Linking/referencing doesn't imply network access. One can be aware of the existence of an identifier, but nothing necessarily beyond that.

Reply:
The idea for this header comes from @timbl. The inclusion for `public` was to mimic Google-docs-like functionality: _what I write here, can it be seen and/or edited by the world?_
The unrestricted linking id indeed part of that.

Definitely nothing to do with `public` having less privileges than `user` (is that even possible in current WAC, BTW?).

Reply:
Ack re `public` use.

The general use case/need was originally documented recorded in https://github.com/linkeddata/dokieli/issues/96 , https://github.com/solid/specification/issues/171 .. (2016). Not the specifics.

Reply:
Yes @bblfish basically the public readability , like the user write frankly, is for modifying the options to users.    If the user can't write it, then the UI will not make the fields editable, will remove action buttons to change things, and suppress blank empty fields.  Maybe grey color text slightly.

If the thing is NOT viewable by the public, then his will suppress social actions which are public about it, like liking it, bookmarking it, "add this to my profile", etc etc, dropping it into a public chat.

If the thing  IS viewable by the public, and editable by the user, then using colors or fonts etc and actual warnings make it clear that anything the user types will be public. "People who can see this: 3 billion".

Those are the biggest uses I can think of ..There are others I am completely missing.

There is no reason that other non-WAC systems can't still just **supply a WAC-Allow header anyway**. Suggest they all be required to forSolid, whatever the ACL language they use. ("Control" would mean "ACP control read and write" for example in ACP